src/main/java/mate/academy/springbootstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package mate.academy.springbootstore.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootstore.dto.order.OrderItemResponseDto;
+import mate.academy.springbootstore.dto.order.OrderRequestDto;
+import mate.academy.springbootstore.dto.order.OrderResponseDto;
+import mate.academy.springbootstore.exception.EntityNotFoundException;
+import mate.academy.springbootstore.mapper.OrderMapper;
+import mate.academy.springbootstore.model.CartItem;
+import mate.academy.springbootstore.model.Order;
+import mate.academy.springbootstore.model.OrderItem;
+import mate.academy.springbootstore.model.ShoppingCart;
+import mate.academy.springbootstore.model.Status;
+import mate.academy.springbootstore.model.User;
+import mate.academy.springbootstore.repository.OrderItemRepository;
+import mate.academy.springbootstore.repository.OrderRepository;
+import mate.academy.springbootstore.repository.ShoppingCartRepository;
+import mate.academy.springbootstore.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository cartRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(OrderRequestDto orderRequestDto, Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        ShoppingCart cart = cartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Cart not found"));
+
+        Set<CartItem> cartItems = cart.getCartItems();
+
+        if (cartItems.isEmpty()) {
+            throw new RuntimeException("Cart is empty");

```

Comment:
provide parameter to message
```suggestion
            throw new OrderProcessingExcpeiton("Cart is empty");
```